apps/framework-cli/src/framework/core/execute.rs

```diff
@@ -122,6 +122,7 @@ pub async fn execute_online_change(
         .await
         .map_err(Box::new)?;
 
+    processes::execute_leader_changes(process_registries, &plan.changes.processes_changes).await?;
```

Comment:
Can you explain this change?
Not sure I understand why this solves the issue

Reply:
Recently, we put blocks behind leadership lock. That change meant `processes::execute_changes` was only updating stuff that can happen on any instance, and there's now `processes:: execute_leader_changes` that gets called in a leadership instance. That change also caused this bug because the file watcher codepath was only calling `processes::execute_changes`.

Reply:
kk makes sense